components/studies/ui/src/main/resources/PhenoTips/StudyBindingClass.xml

```diff
@@ -622,28 +634,40 @@ $xwiki.ssx.use($className)
 {{/velocity}}
 
 {{velocity}}
-#set ($toProcess = $collectionstool.queue)
-#set ($discard = $toProcess.add($xcontext.userReference.toString()))
-#set ($groups = [])
-#set ($studies = $collectionstool.sortedSet)
-#foreach ($iteration in [0..1000])
-  #if ($toProcess.empty)
-    #break
+#set($studiesVisibilityPreference=$xwiki.getDocument('PhenoTips.StudyClass').getObject('PhenoTips.StudyVisibilityClass').getProperty('visibility').value)
+#if ($studiesVisibilityPreference == 'open')
+  #set ($studyClassName = 'PhenoTips.StudyClass')
+  #set ($studyTemplateName = 'PhenoTips.StudyTemplate')
+  #set ($query = $services.query.xwql("from doc.object(${studyClassName}) as std where doc.fullName &lt;&gt; :template").bindValue('template', $studyTemplateName))
+  #set ($studyDocs = $xwiki.wrapDocs($query.execute()))
+  #set ($studies = $collectionstool.sortedSet)
+  #foreach ($study in $studyDocs)
+    #set ($discard = $studies.add("$study"))
   #end
-  #set ($item = $toProcess.poll())
-  #foreach ($group in $xwiki.wrapDocs($xwiki.rightsmanager.getAllGroupsNamesForMember($item)))
-    #set ($groupObj = $group.getObject('PhenoTips.PhenoTipsGroupClass'))
-    #if (!$groupObj &amp;&amp; $group.documentReference.name.endsWith(' Administrators'))
-      #set ($groupObj = $xwiki.getDocument($services.model.resolveDocument($stringtool.removeEnd($group.documentReference.name, ' Administrators'), $group.documentReference.parent)).getObject('PhenoTips.PhenoTipsGroupClass'))
+#else
+  #set ($toProcess = $collectionstool.queue)
+  #set ($discard = $toProcess.add($xcontext.userReference.toString()))
+  #set ($groups = [])
+  #set ($studies = [])
```

Comment:
Here the order in which the studies are listed is important, so leave it as:

    #set ($studies = $collectionstool.sortedSet)